### ISSUE TITLE
Make CaptureController an EventTarget

### ISSUE THREAD

**Issue by fred-wang** [screen-share/captured-mouse-events#9]
See https://dom.spec.whatwg.org/#eventtarget

The extension in the spec is:

```
partial interface CaptureController {
  attribute EventHandler oncapturedmousechange;
};
```

I see interfaces with similar interfaces with similar `onsomething` EventHandler attribute derive from `EventTarget` but not sure whether that's something that is a requirement or even just that we consider interesting here.

**Comment by eladalon1983** [comment]
Are you aware of precedents of an extension spec introducing inheritance?

**Comment by fred-wang** [comment]
> Are you aware of precedents of an extension spec introducing inheritance?

That does not seem possible:

"Note: A partial interface definition cannot specify that the interface [inherits](https://webidl.spec.whatwg.org/#dfn-inherit) from another [interface](https://webidl.spec.whatwg.org/#dfn-interface). Inheritance is to be specified on the original interface definition."

https://webidl.spec.whatwg.org/#dfn-partial-interface

**Comment by fred-wang** [comment]
@eladalon1983 For now the solutions I have in mind are:

1) Get this inheritance change accepted in the Screen Capture directly.
2) Introduce a separate class deriving from `EventTarget` and exposing `on*`  events. This class would be exposed as a new attribute of `CaptureController`, or alternatively it could inherit from `CaptureController` and passed as an option to `getDisplayMedia` (but not sure this would work).

**Comment by eladalon1983** [comment]
Short-term, let's redefine CaptureController in our spec as inheriting from EventTarget. (That is, not use `partial`, but say that we're redefining it outright.) Will you send a PR to that effect?

Long-term, let's try to get this accepted in the Screen Capture spec. I'll start the discussion. (Edit: I've filed w3c/mediacapture-screen-share/issues/268 and w3c/mediacapture-screen-share/issues/269 and asked for a time slot to discuss this during the June interim of the WebRTC WG, which manages the spec where CaptureController is defined.)

**Comment by fred-wang** [comment]
@eladalon1983 OK, I submitted a first PR here: https://github.com/screen-share/mouse-events/pull/10 ; will work on the constructor stuff in a separate PR

**Comment by eladalon1983** [comment]
Frédéric, do you mind if I close this issue now?

**Comment by fred-wang** [comment]
@eladalon1983 OK when https://github.com/screen-share/captured-mouse-events/pull/10 is merged :-)  and then we can close issue #1 too

**Comment by fred-wang** [comment]
Oh nevermind, we actually don't need to specify inheritance on partial interfaces. I already had removed that from PR #10...
